### PR TITLE
ENH: Add a function to filter inactive phases

### LINKED ISSUE #100: Filter inactive phases in calculate and equilibrium
Phases containing sublattices with all inactive components are currently handled differently in calculate and equilibrium. In equilibrium a DoFError is raised and in calculate, the DoFError is caught and the phase is suspended with a warning.

Options:
1. Make equilibrium catch the error and give a similar warning (kind of ugly for higher-order systems with many phases)
2. Filter out phases on the front ends and just raise an error
    a. either with just some code
    b. provide a database convince function that gives active phases given a compositions

A sample piece of code for (currently) finding active phases looks like:
```python
active_phases = [phase for phase in dbf.phases.keys() if all(
                 len(set(comps).intersection(subl)) > 0 for subl in dbf.phases[phase].constituents)] 
```

Architecture for future species support (how primitive components of species are represented) may change the implementation of this.

### LINKED ISSUE #129: more descriptive error messages, if possible
When inputting a phase into the phase list that does not exist in the TDB, attempting to use this list in, e.g., binplot() results in the not-terribly-useful error message: KeyError: <wrong phase>.  It's fairly easy to figure out, but it would be nice if there were a check or a message like "this phase <wrong phase> does not exist in the TDB."

If I come across additional places for more helpful messages, I'll update this until it's closed.

Update: Also, there should be an error message for the bug in issue #116, instead of silently calculating the wrong free energy.

### PR DESCRIPTION
**Pull request by bocklund**
Closes #100 and closes #129

I use this all the time. @richardotis what do you think about including this in pycalphad proper?

Any thoughts on doing this automatically in equilibrium calculations? That way users could always pass in phases=dbf.phases.keys() without getting any error messages. In large databases, your only other choice is to tediously select phases in the system you want to calculate.

### CODE REVIEW

**Comment by bocklund** [comment]
Having users do this (by updating examples) and applying this closes #129

**Comment by richardotis** [comment]
This sounds like a good idea. Adding this to `equilibrium` (but not `calculate`) would be useful.

**Comment by richardotis** [comment]
If the result of `filter_phases` is empty, we should give the user a nice error message like "No phase with the specified components could be found in the database."
The user probably made a typo somewhere, so the error message should give a hint to check the component list and/or the database.

Semi-related comment: I would be okay with creating a base `EquilibriumError` exception class and then basing all "logical" errors (as opposed to errors from bugs) returning that exception or a subclass of it.

File: pycalphad/core/equilibrium.py [hunk 1]
```diff
@@ -193,7 +193,14 @@ def equilibrium(dbf, comps, phases, conditions, output=None, model=None,
...
     from pycalphad import __version__ as pycalphad_version
-    active_phases = unpack_phases(phases) or sorted(dbf.phases.keys())
+    phases = unpack_phases(phases) or sorted(dbf.phases.keys())
+    # remove phases that cannot be active
+    possible_active_phases = filter_phases(dbf, comps)
+    active_phases = sorted(set(possible_active_phases).intersection(set(phases)))
+    if len(possible_active_phases) == 0:
+        raise ConditionError('There are no phases in the Database that can be active with components {0}'.format(comps))
+    if len(active_phases) == 0:
+        raise ConditionError('None of the passed phases ({0}) are active. List of active phases: {1}.'.format(phases, possible_active_phases))
```
Comment (richardotis) [inline pycalphad/core/equilibrium.py]: Should be "List of possible phases"

File: pycalphad/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+The utils test module contains tests for pycalphad utilities.
+"""
+
+from pycalphad import Database
+from pycalphad.core.utils import filter_phases
+
+from pycalphad.tests.datasets import ALNIPT_TDB
+
+ALNIPT_DBF = Database(ALNIPT_TDB)
+
```
Comment (richardotis) [inline pycalphad/tests/test_utils.py]: Needs an integration test to show the equilibrium logic is working. Ideally for both ConditionErrors

**Pull request merged**